src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts

```diff
@@ -808,78 +808,93 @@ g.test('depth_stencil_attachment,sample_counts_mismatch')
     }
   });
 
-g.test('depth_stencil_attachment')
+g.test('depth_stencil_attachment,loadOp_storeOp_match_depthReadOnly_stencilReadOnly')
   .desc(
     `
   Test GPURenderPassDepthStencilAttachment Usage:
     - depthReadOnly and stencilReadOnly must match if the format is a combined depth-stencil format.
-    - depthLoadOp and depthStoreOp must be provided iff the format has a depth aspect and
-      depthReadOnly is not true.
-    - stencilLoadOp and stencilStoreOp must be provided iff the format has a stencil aspect and
-      stencilReadOnly is not true.
+    - depthLoadOp and depthStoreOp must be provided if the format has a depth aspect and
+      depthReadOnly is not true and must not be provided if it depthReadOnly is true.
+    - stencilLoadOp and stencilStoreOp must be provided if the format has a stencil aspect and
+      stencilReadOnly is not true and must not be provided if stencilReadOnly is true.
   `
   )
   .params(u =>
-    u //
+    u
+      .beginSubcases() // Note: It's easier to debug if you comment this line out as you can then run an individual case.
       .combine('format', kDepthStencilFormats)
-      .beginSubcases()
-      .combine('depthReadOnly', [false, true])
-      .combine('stencilReadOnly', [false, true])
-      .combine('setDepthLoadStoreOp', [false, true])
-      .combine('setStencilLoadStoreOp', [false, true])
+      .combine('depthReadOnly', [undefined, true, false])
+      .combine('depthLoadOp', [undefined, 'clear', 'load'] as GPULoadOp[])
+      .combine('depthStoreOp', [undefined, 'discard', 'store'] as GPUStoreOp[])
+      .combine('stencilReadOnly', [undefined, true, false])
+      .combine('stencilLoadOp', [undefined, 'clear', 'load'] as GPULoadOp[])
+      .combine('stencilStoreOp', [undefined, 'discard', 'store'] as GPUStoreOp[])
   )
-  .beforeAllSubcases(t => {
-    t.selectDeviceForTextureFormatOrSkipTestCase(t.params.format);
-  })
   .fn(t => {
     const {
       format,
       depthReadOnly,
+      depthLoadOp,
+      depthStoreOp,
       stencilReadOnly,
-      setDepthLoadStoreOp,
-      setStencilLoadStoreOp,
+      stencilLoadOp,
+      stencilStoreOp,
     } = t.params;
 
-    let isValid = true;
-    const info = kTextureFormatInfo[format];
-    if (info.depth && info.stencil) {
-      isValid &&= depthReadOnly === stencilReadOnly;
+    if (format === 'depth32float-stencil8' && !t.device.features.has('depth32float-stencil8')) {
```

Comment:
I think the general pattern we use for this kind of thing is:
```
const info = kTextureFormatInfo[format];
t.selectDeviceOrSkipTestCase(info.feature);
```

Reply:
done

Reply:
I think we should be able to remove this if clause with the change.

Reply:
sorry I missed that. done